uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java

```diff
@@ -232,16 +280,40 @@ private TLSSettings buildTlsSettings(IotThingSpec thingSpec, String brokerId) {
                           .build();
     }
 
-    @SuppressWarnings("PMD.UnusedFormalParameter")
+    private void processDiscoveryResponse(String brokerId, DiscoverResponse response) {
+        if (response.getGGGroups() != null) {
+            final Optional<GGGroup> groupOpt = response.getGGGroups()
+                                                       .stream()
+                                                       .findFirst();
+            if (groupOpt.isPresent()) {
+                final GGGroup group = groupOpt.get();
+                final String ca = group.getCAs()
+                                       .get(0);
+                putBrokerCa(brokerId, ca);
+                final ConnectivityInfo connInfo = group.getCores()
+                                                       .get(0)
+                                                       .getConnectivity()
+                                                       .get(0);
```

Comment:
this will return more than one option, this must be handled. Specifically, try connecting to each host and port pair until we're able to connect.

Reply:
Fixed bcb91db97897bdeec075453a3508f8c01a14f20e